src/main/kotlin/com/bantolomeus/service/BookService.kt

```diff
@@ -16,84 +16,87 @@ class BookService(private val bookRepository: BookRepository,
 
     fun createBook(bookDTO: BookDTO): BookDTO {
         val savedBookDTO = bookRepository.saveBookIfItNotExists(bookDTO)
-        if (savedBookDTO.currentPage > 0L) {
+        if (savedBookDTO.currentPage > 0) {
             saveBookUpdate(savedBookDTO)
         }
         return bookDTO
     }
 
     fun updateBook(bookUpdate: BookUpdateInputDTO, bookName: String): BookUpdatesFileDTO {
-        var response = BookUpdatesFileDTO()
-        var books = bookRepository.getBooks().books.filter { it.name != bookName }
-        var oldPage = Long.MAX_VALUE
-        val foundBook = bookRepository.getBookByName(bookName)
+        val foundBook = bookRepository.getBookByName(bookName) ?: return BookUpdatesFileDTO()
 
-        if (foundBook.name == bookName) {
-            foundBook.let {
-                oldPage = it.currentPage
-                it.currentPage = bookUpdate.currentPage
-                if (it.currentPage >= it.pagesTotal) {
-                    it.readTime = (Date().time - dateFormat.parse(it.dateStarted).time) / DIVISOR_FOR_DAY
-                }
-            }
-            books = mutableListOf(foundBook) + books.toMutableList()
+        if (foundBook.currentPage == foundBook.pagesTotal) {
+            foundBook.readTime = ((Date().time - dateFormat.parse(foundBook.dateStarted).time) / DIVISOR_FOR_DAY) + 1
+        }
+        val oldPage = foundBook.currentPage
+        foundBook.currentPage = bookUpdate.currentPage
+        val pagesRead = bookUpdate.currentPage.minus(oldPage)
+        if (pagesRead <= 0 || oldPage >= bookUpdate.currentPage || bookUpdate.currentPage > foundBook.pagesTotal) {
+            return BookUpdatesFileDTO()
         }
 
-        if (oldPage < bookUpdate.currentPage && foundBook.name == bookName
-                && bookUpdate.currentPage <= foundBook.pagesTotal) {
-            bookRepository.saveBooks(BooksFileDTO(books.toMutableList()))
-            val bookUpdates = bookUpdatesRepository.getBookUpdates()
-            val currentDate = dateFormat.format(Date())
-            val foundBookUpdate = bookUpdates.bookUpdates.filter { it.date == currentDate && it.name == bookName }
-                    .getOrElse(0) { _ -> BookUpdateOutputDTO()}
+        // todo this should happen in the repository I think.
+        //      repository.saveBook(foundBook) should encapsulate the filter stuff
```

Comment:
true

Reply:
@Husterknupp why do I not see the comments in the current branch but here? That makes no sense.

Reply:
Why are these changes not visible in the branch?

Reply:
@Husterknupp did you pushed it on ```refactor/payload-and-url-refactoring#79#81#90``` or on ```refactor/payload-and-url-refactoring/#79/#81/#90``` ?
Can you look it up in your git history?

Reply:
I see your commit in the git log...

Reply:
Solved, I restarted Intellij and then I could see the comments 👍

Reply:
I implemented a method into the repository for you. I do not like the idea of methods that handle a lot of cases thats why I wrote a method for this special case to reduce the complexity and increase the readability.

Reply:
I see the method. But maybe finalize it: 
1) As discussed, please create a method `updateBook(BookDTO)` which replaces a book in the repo file. All filtering should happen within that repo method, not in the service
2) why do you love `MutableList` s so much? 

BookRepository#getAllBooksExcept
```
objectMapper.readValue<List<BookDTO>>(File(bookFile)).filter { it.name != bookName }
```

does the same thing as the version with the MutableList. Why do you need it?

Reply:
Done. I never said I love it. I just used it and never changed it.

Reply:
#sarcasm ;-) Because you were using it more than once and it's actually more text to write for no benefit. So love was the only reasonable explanation :-p